refactor(transformer): move implementation of ArrowFunction to common/ArrowFunctionConverter

Dunqing opened the pull request:
Part of #7074

In order can reuse the ability of the `ArrowFunction` plugin, we moved out the implementation to common, then we can use use in other plugins

Dunqing commented:
* **#7107** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7107?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Dunqing and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

overlookmotel reviewed (approved):
You got rid of the `RefCell`. Nice!

overlookmotel commented:
### Merge activity

* **Nov 4, 10:07 AM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel commented:
Oh damn it! @Dunqing merge conflicts with your other PR.

Dunqing committed:
refactor(transformer): move implementation of ArrowFunction to common/ArrowFunctionConverter (#7107)

Part of #7074

In order can reuse the ability of the `ArrowFunction` plugin, we moved out the implementation to common, then we can use use in other plugins
